src/apps/sources/pages/ViewSourcePage.tsx

```diff
@@ -97,6 +110,15 @@ export const ViewSourcePage: React.FC<Props> = ({
             />
           </Grid>
         </Grid>
+        {!showEditButton ? null : (
+            <Link to={`${url}edit/`}>
+              <Tooltip title="Edit this Source">
+                <Fab color="primary" className="fab">
+                  <EditIcon />
+                </Fab>
+              </Tooltip>
+            </Link>
+        )}
```

Comment:
This is also common for Source and Dictionary. We can move it to Containers.

Reply:
Made necessary changes please check